From a138455bd35a7f3ccfb77a6eb309733c43a229fd Mon Sep 17 00:00:00 2001
From: Krishna Manikandan <mkrishn@codeaurora.org>
Date: Fri, 30 Jun 2017 11:49:47 +0530
Subject: msm: mdss: Add check for fence count

Add a check to ensure that the acquire fence count
does not exceed the maximum possible value.

Change-Id: I7198899be2d720214152d49fdbb6b6a69750fb3a
Signed-off-by: Krishna Manikandan <mkrishn@codeaurora.org>
---
 drivers/video/msm/mdss/mdss_mdp_layer.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/video/msm/mdss/mdss_mdp_layer.c b/drivers/video/msm/mdss/mdss_mdp_layer.c
index 6214afc..3ff446c 100644
--- a/drivers/video/msm/mdss/mdss_mdp_layer.c
+++ b/drivers/video/msm/mdss/mdss_mdp_layer.c
@@ -2234,6 +2234,14 @@ int mdss_mdp_layer_pre_commit_wfd(struct msm_fb_data_type *mfd,
 		sync_pt_data = &mfd->mdp_sync_pt_data;
 		mutex_lock(&sync_pt_data->sync_mutex);
 		count = sync_pt_data->acq_fen_cnt;
+
+		if (count >= MDP_MAX_FENCE_FD) {
+			pr_err("Reached maximum possible value for fence count\n");
+			mutex_unlock(&sync_pt_data->sync_mutex);
+			rc = -EINVAL;
+			goto input_layer_err;
+		}
+
 		sync_pt_data->acq_fen[count] = fence;
 		sync_pt_data->acq_fen_cnt++;
 		mutex_unlock(&sync_pt_data->sync_mutex);
-- 
cgit v1.1

